crates/connectors/ndc-postgres/src/convert.rs

```diff
@@ -0,0 +1,117 @@
+//! Functions to convert between internal error types and the error types exposed by ndc-sdk.
+
+use ndc_sdk::connector;
+
+pub(crate) fn execution_error_to_query_error(
```

Comment:
Why pub(crate) and not just pub? I'm assuming that it could be useful to use these functions outside this crate, and there's no harm in that either. I think we can default to simple pub.

Reply:
And could we please add docstrings to these functions as well? :)

Reply:
All very helpful suggestions. Thanks! I have done them all.